omnigibson/simulator.py

```diff
@@ -505,9 +508,12 @@ def play(self):
                 for obj in self._objects_to_initialize:
                     obj.initialize()
                 self._objects_to_initialize = []
-                # Also update the scene registry
-                # TODO: A better place to put this perhaps?
-                self._scene.object_registry.update(keys="root_handle")
+
+            # If we were stopped, take an additional sim step to make sure simulator is functioning properly
+            # We need to do this because for some reason omniverse exhibits strange behavior if we do certain operations
+            # immediately after playing; e.g.: syncing USD poses when flatcache is enabled
+            if was_stopped:
+                self.step()
```

Comment:
hmm another physics leakage. Is it absolutely needed? would self.render() suffice here?

Reply:
no